dotCMS/src/main/java/com/dotcms/rekognition/actionlet/RekognitionActionlet.java

```diff
@@ -0,0 +1,154 @@
+            //Check if there is already a Tag with the name TAGGED_BY_AWS, if there is Tags were already generated and nothing to do here
+            if (tags.stream().anyMatch(tag -> TAGGED_BY_AWS.equalsIgnoreCase(tag.getTagName()))) {
+                Logger.warn(this,"Tags already generated");
+                return;
+            }
+
+            imageOpt = fields.stream().filter(BinaryField.class::isInstance)
+                    .map(field -> Try.of(()->contentlet.getBinary(field.variable())).getOrNull())
+                    .filter(img -> null != img && UtilMethods.isImage(img.getAbsolutePath()))
+                    .findFirst();
+
+            //Check if there is a Binary Field and if there is an Image set in it, if not there is nothing to do here
+            if (!imageOpt.isPresent()) {
+                Logger.warn(this,"There is no Binary Field or an Image is not set in it");
+                return;
+            }
+            image = imageOpt.get();
+
+            //Get Values from Secrets
+            final Host host = Try.of(() -> APILocator.getHostAPI().find(contentlet.getHost(), APILocator.systemUser(),false)).getOrElse(APILocator.systemHost());
+            appSecrets = APILocator.getAppsAPI().getSecrets(AMAZON_REKOGNITION_APP_CONFIG_KEY,true,host,APILocator.systemUser());
+
+            if(!appSecrets.isPresent()) {
+                Logger.warn(RekognitionActionlet.class,"There is no config set, please set it via Apps Tool");
+                return;
+            }
+
+            final String accessKey = appSecrets.get().getSecrets().get(ACCESS_KEY_VAR).getString();
+            final String secretAccessKey = appSecrets.get().getSecrets().get(SECRET_ACCESS_KEY_VAR).getString();
+            final int maxLabels = Integer.parseInt(appSecrets.get().getSecrets().get(MAX_LABELS_VAR).getString());
+            final float minConfidence = Float.parseFloat(appSecrets.get().getSecrets().get(MIN_CONFIDENCE_VAR).getString());
+            //End Get Values from Secrets
+
+            if (image.length() > IMAGE_MAX_LENGTH) {
+
+                final Map<String, String[]> args = new HashMap<>();
+                args.put("resize_w", new String[]{"1000"});
+                image = new ResizeImageFilter().runFilter(image, args);
+            }
+
+            final List<String> awsTags = new RekognitionAPI(accessKey,secretAccessKey).generateTags(image, maxLabels, minConfidence);
+
+            Logger.debug(this,"Tags generated by AWS: " + awsTags.toString());
+
+            awsTags.add(TAGGED_BY_AWS);
+            for (final String tag : awsTags) {
+                tagAPI.addContentletTagInode(tag, contentlet.getInode(), contentlet.getHost(), tagFieldOpt.get().variable());
+            }
+
+            HibernateUtil.addCommitListener(contentlet.getInode(),
+                    ()->this.refresh(contentlet));
+
+            CacheLocator.getContentletCache().remove(contentlet);
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6017382)